test/unittests/test_loader.cpp

```diff
@@ -328,3 +321,250 @@ TEST_F(loader, load_and_create_abi_mismatch)
+
+    vm = evmc_load_and_configure("path,", &ec);
+    EXPECT_TRUE(vm);
+    EXPECT_TRUE(recorded_options.empty());
+    EXPECT_EQ(ec, EVMC_LOADER_SUCCESS);
+}
+
+TEST_F(loader, load_and_configure_single_option)
+{
+    supported_options["o"] = "1";
+    supported_options["O"] = "2";
+
+    setup("path", "evmc_create", create_vm_with_set_option);
+
+    evmc_loader_error_code ec;
+    auto vm = evmc_load_and_configure("path,o=1", &ec);
+    EXPECT_TRUE(vm);
+    ASSERT_EQ(recorded_options.size(), 1);
+    EXPECT_EQ(recorded_options[0].first, "o");
+    EXPECT_EQ(recorded_options[0].second, "1");
+    EXPECT_EQ(ec, EVMC_LOADER_SUCCESS);
+
+    recorded_options.clear();
+    vm = evmc_load_and_configure("path,O=2", &ec);
+    EXPECT_TRUE(vm);
+    ASSERT_EQ(recorded_options.size(), 1);
+    EXPECT_EQ(recorded_options[0].first, "O");
+    EXPECT_EQ(recorded_options[0].second, "2");
+    EXPECT_EQ(ec, EVMC_LOADER_SUCCESS);
+}
+
+TEST_F(loader, load_and_configure_uknown_option)
+{
+    supported_options["x"] = "1";
+
+    setup("path", "evmc_create", create_vm_with_set_option);
+
+    evmc_loader_error_code ec;
+    auto vm = evmc_load_and_configure("path,z=1", &ec);
+    EXPECT_FALSE(vm);
+    ASSERT_EQ(recorded_options.size(), 1);
+    EXPECT_EQ(recorded_options[0].first, "z");
+    EXPECT_EQ(recorded_options[0].second, "1");
+    EXPECT_EQ(ec, EVMC_LOADER_INVALID_OPTION_NAME);
+    EXPECT_STREQ(evmc_last_error_msg(), "vm_with_set_option (path): unknown option 'z'");
+    EXPECT_EQ(destroy_count, create_count);
+
+    recorded_options.clear();
+    vm = evmc_load_and_configure("path,x=2,", &ec);
+    EXPECT_FALSE(vm);
+    ASSERT_EQ(recorded_options.size(), 1);
+    EXPECT_EQ(recorded_options[0].first, "x");
+    EXPECT_EQ(recorded_options[0].second, "2");
+    EXPECT_EQ(ec, EVMC_LOADER_INVALID_OPTION_VALUE);
+    EXPECT_STREQ(evmc_last_error_msg(),
+                 "vm_with_set_option (path): unsupported value '2' for option 'x'");
+    EXPECT_EQ(destroy_count, create_count);
+}
+
+TEST_F(loader, load_and_configure_multiple_options)
+{
+    supported_options["a"] = "_a";
+    supported_options["b"] = "_b";
+    supported_options["c"] = "_c";
+
+    setup("path", "evmc_create", create_vm_with_set_option);
+
+    evmc_loader_error_code ec;
+    auto vm = evmc_load_and_configure("path,a=_a,b=_b,c=_c,b=_b", &ec);
+    EXPECT_TRUE(vm);
+    ASSERT_EQ(recorded_options.size(), 4);
+    EXPECT_EQ(recorded_options[0].first, "a");
+    EXPECT_EQ(recorded_options[0].second, "_a");
+    EXPECT_EQ(recorded_options[1].first, "b");
+    EXPECT_EQ(recorded_options[1].second, "_b");
+    EXPECT_EQ(recorded_options[2].first, "c");
+    EXPECT_EQ(recorded_options[2].second, "_c");
+    EXPECT_EQ(recorded_options[3].first, "b");
+    EXPECT_EQ(recorded_options[3].second, "_b");
+    EXPECT_EQ(ec, EVMC_LOADER_SUCCESS);
+
+    recorded_options.clear();
+    vm = evmc_load_and_configure("path,a=_a,b=_b,a=_a,", &ec);
```

Comment:
The last instance of the same option overrides the earlier ones. Can you test something like `a=_a,b=_b,a=_c` and ensure that `a` == `_c` ?

Also this probably should be documented.

Reply:
Done.